os/kernel/log_dump/log_dump.c

```diff
@@ -228,7 +228,7 @@ int log_dump_compress_lastblock(void)
 
 	last_comp_block_size = CONFIG_LOG_DUMP_CHUNK_SIZE;
 
-	compress_ret = compress_block(&last_comp_block[4], &last_comp_block_size, uncomp_buf[uncomp_idx], uncomp_curbytes);
+	compress_ret = compress_block(&last_comp_block[4], (long unsigned int *)&last_comp_block_size,  (unsigned char *)uncomp_buf[uncomp_idx], uncomp_curbytes);
```

Comment:
why cant we update variable definitions in the log_dump according to the compression module?. Typecasting needs to carefully used without any side effects.